cycle-app.py

```diff
@@ -0,0 +1,270 @@
+                id='dark-toggle',
+                type='button',
+                n_clicks=0),
+    html.Div(id='plot-bin',
+             children=[html.Img(src='',
+                                alt='May take a moment to load...',
+                                id='geometry-plot',),
+                       html.Img(src='',
+                                alt='May take a moment to load...',
+                                id='eigen-plot')]),
+])
+
+# Populates wheel-table parameter with data
+
+
+@app.callback(Output('wheel-table', 'data'), [Input('bike-dropdown', 'value'), Input('reset-button', 'n_clicks')])
+def populate_wheel_data(value, n_clicks):
+    parPure = new_par(value)
+    data = []
+    labels = []
+    fW = []
+    rW = []
+    for i in WHEEL_LABELS:
+        labels.append({'label': i})
+    for i in range(8):
+        if i < 4:
+            fW.append({'fW': parPure.get(pList[i])})
+        else:
+            rW.append({'rW': parPure.get(pList[i])})
+    for c, d, e in zip(labels, fW, rW):
+        zipped = {}
+        zipped.update(c)
+        zipped.update(d)
+        zipped.update(e)
+        data.append(zipped)
+    return data
+
+    # Populates frame-table parameter with data
+
+
+@app.callback(Output('frame-table', 'data'), [Input('bike-dropdown', 'value'), Input('reset-button', 'n_clicks')])
+def populate_frame_data(value, n_clicks):
+    parPure = new_par(value)
+    data = []
+    labels = []
+    rB = []
+    fA = []
+    for i in FRAME_LABELS:
+        labels.append({'label': i})
+    for i in range(12, len(pList)):
+        if i < 19:
+            rB.append({'rB': parPure.get(pList[i])})
+        else:
+            fA.append({'fA': parPure.get(pList[i])})
+    for c, d, e in zip(labels, rB, fA):
+        zipped = {}
+        zipped.update(c)
+        zipped.update(d)
+        zipped.update(e)
+        data.append(zipped)
+    return data
+
+    # Populates general-table parameter with data
+
+
+@app.callback(Output('general-table', 'data'), [Input('bike-dropdown', 'value'), Input('reset-button', 'n_clicks')])
+def populate_general_data(value, n_clicks):
+    parPure = new_par(value)
+    data = []
+    labels = []
+    con = []
+    for i in GENERAL_LABELS:
+        labels.append({'label': i})
+    for i in range(8, 12):
+        con.append({'con': parPure.get(pList[i])})
+    for c, d in zip(labels, con):
+        zipped = {}
+        zipped.update(c)
+        zipped.update(d)
+        data.append(zipped)
+    return data
+
+    # Updates geo-plot & eigen-plot path with Dropdown value or edited DataTable values
+
+
+@app.callback([Output('geometry-plot', 'src'),
+               Output('eigen-plot', 'src')],
+              [Input('bike-dropdown', 'value'),
+               Input('wheel-table', 'data'),
+               Input('frame-table', 'data'),
+               Input('general-table', 'data')])
+def plot_update(value, x, y, z):
+    ctx = dash.callback_context
+    speeds = np.linspace(0, 10, num=100)
```

Comment:
It would be nice if the user could specify the minimum and maximum speeds via the GUI. If you do that, then the num parameter should be calculated such that there is data point every 0.1 m/s or so.

Reply:
Sure. We'll put that in the works.